### PR TITLE
ansible 2.10.3

### PR DESCRIPTION
**Pull request by domrim**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

install process is a bit hacky but worked and symlinks all needed bins.

### CODE REVIEW

**Review by samford — DISMISSED** [review]
The resource updates here are correct, outside of the `contextlib2`/`cryptography` issue in the comment below.

It would be good if someone who uses Ansible could comment on the move from 2.9.x to 2.10.x, considering that https://releases.ansible.com/ansible/ doesn't list any 2.10.x archives and the latest 2.9.x archive (2.9.15) was released pretty recently (2020-11-03).

Despite my experience in updating this formula, I don't use Ansible and can't say whether this version bump will impact users in some way. We have an `ansible@2.8` formula, so it may be appropriate to create an `ansible@2.9` formula if this update leads to breaking changes for users. The [Ansible 2.10 Porting Guide](https://docs.ansible.com/ansible/devel/porting_guides/porting_guide_2.10.html) and the [Changelog for 2.10](https://github.com/ansible-community/ansible-build-data/blob/main/2.10/CHANGELOG-v2.10.rst) may help someone with knowledge of Ansible to understand the situation.

I can't comment on the `virtualenv` changes but when I built this locally and compared it to the contents of the 2.9.15 bottle, it looked reasonable to me. Someone with more experience working with Python formulae may have some feedback but it seems to get the job done.

@Homebrew/core What do we think about bumping this to 2.10.x? Should we create an `ansible@2.9` formula (like we have `ansible@2.8`)?

---

Instead of removing the `livecheck` block, we should replace it with:

```ruby
livecheck do
  url :stable
end
```

This will use the `Pypi` strategy on the `stable` URL, so the check aligns with the `stable` source. Without this, livecheck would check `head` (the GitHub Git tags) by default, which isn't as preferable.

**Comment by domrim** [comment]
@samford 
I use Ansible on an day to day basis at work so i can provide some feedback for the upgrade.
This update changes a lot of the internals of Ansible but doesn't change its behaviour for the user (aside the actual functional changes) because we build ansible with all collections included. I use all my playbooks / roles (originally written for Ansible 2.9) unchanged with this 2.10.3 build without any problem, so i think we don't need an ansible@2.9 formula.

File: Formula/ansible.rb [hunk 1]
```diff
@@ -178,23 +178,23 @@ class Ansible < Formula
...
     url "https://files.pythonhosted.org/packages/1f/bb/5d3246097ab77fa083a61bd8d3d527b7ae063c7d8e8671b1cf8c4ec10cbe/colorama-0.4.4.tar.gz"
     sha256 "5941b2b48a20143d2267e95b1c2a7603ce057ee39fd88e7329b0c292aa16869b"
   end
 
-  resource "contextlib2" do
-    url "https://files.pythonhosted.org/packages/02/54/669207eb72e3d8ae8b38aa1f0703ee87a0e9f88f30d3c0a47bebdb6de242/contextlib2-0.6.0.post1.tar.gz"
-    sha256 "01f490098c18b19d2bd5bb5dc445b2054d2fa97f09a4280ba2c5f3c394c8162e"
+  resource "cryptography" do
+    url "https://files.pythonhosted.org/packages/94/5c/42de91c7fbdb817b2d9a4e64b067946eb38a4eb36c1a09c96c87a0f86a82/cryptography-3.2.1.tar.gz"
+    sha256 "d3d5e10be0cf2a12214ddee45c6bd203dab435e3d83b4560c03066eda600bfe3"
```
[the remark this answers is not in the document]
Comment (domrim) [reply Formula/ansible.rb]: fixed it and rebased again

**Review by samford — COMMENTED** [review]
This looks good to me now and I think it's probably ready to merge once other maintainers have a moment to take a look and provide their thoughts. If that doesn't seem likely after 24 hours or so, I'll go ahead and approve this at that time (so it gets merged).

Thanks for your work here, @domrim!

**Comment by Rylan12** [comment]
> It would be good if someone who uses Ansible could comment on the move from 2.9.x to 2.10.x, considering that https://releases.ansible.com/ansible/ doesn't list any 2.10.x archives and the latest 2.9.x archive (2.9.15) was released pretty recently (2020-11-03).

Given that they maintain a 2.8 branch, it's not super surprising to me that they're continuing to make new 2.9 releases. It is a little concerning to me, though, that the 2.10 version doesn't show up on the release page. It makes me think we should add an `ansible@2.9` formulae (see more comments below).

> I can't comment on the `virtualenv` changes but when I built this locally and compared it to the contents of the 2.9.15 bottle, it looked reasonable to me. Someone with more experience working with Python formulae may have some feedback but it seems to get the job done.

`virtualenv` changes look good to me.

> @Homebrew/core What do we think about bumping this to 2.10.x? Should we create an `ansible@2.9` formula (like we have `ansible@2.8`)?

I'm leaning toward adding the 2.9 formula. Ansible is one of our most downloaded formulae (ranked number 43 based on [yearly installs](https://formulae.brew.sh/analytics/install-on-request/365d/)) with over 400,000 yearly installs. `ansible@2.8`, while much farther down the list (number 1152) is still relatively popular with over 3,000 yearly installs. This tells me that older versions of ansible are still desired and would be used. I would guess that upgrading to 2.10 would leave many users stranded in between 2.8 and 2.10. However, I don't think this needs to be a blocker in this PR. It can be added separately (or we can wait until we get a complain that 2.9 is gone).

I'm also a little concerned that I don't see any obvious indication that 2.10 has been released on the ansible website or in the docs so I don't feel confident that all users are aware that 2.10 has been released. Because of that, there may be people who aren't expecting an upgrade and may still need 2.9.

---

**TL;DR:** ✅ from me (including virutalenv changes). I think we should add an `ansible@2.9` formula but can be in a follow up PR.

**Comment by SMillerDev** [comment]
If someone realizes they need an older ansible version they can make a PR to add it. We shouldn't proactively add versioned formulae.

**Comment by domrim** [comment]
> I'm also a little concerned that I don't see any obvious indication that 2.10 has been released on the ansible website or in the docs so I don't feel confident that all users are aware that 2.10 has been released. Because of that, there may be people who aren't expecting an upgrade and may still need 2.9.

The [Roadmap](https://docs.ansible.com/ansible/latest/roadmap/COLLECTIONS_2_10.html) in the Ansible Doc shows the 2.10 release schedule. Also on https://docs.ansible.com/ the "latest" flag links to the 2.10 documentation.

**Review by fxcoudert — DISMISSED** [review]

**Review by SMillerDev — DISMISSED** [review]
Thanks @domrim ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

File: Formula/ansible.rb [hunk 2]
```diff
@@ -682,8 +696,6 @@ def install
     Pathname.glob(libexec/"lib/python*/site-packages/prettytable-0.7.2-py*.egg-info").each do |prettytable_path|
       chmod_R("a+r", prettytable_path)
     end
-
-    man1.install Dir["docs/man/man1/*.1"]
```
Comment (mpolden) [inline Formula/ansible.rb:686]: Why are man pages no longer installed? I've been using Ansible through Homebrew for years, but suddenly `man ansible-playbook` no longer works.

Comment (Rylan12) [reply Formula/ansible.rb:686]: Fix is in https://github.com/Homebrew/homebrew-core/pull/65365
